src/pathplanning.org

```diff
@@ -0,0 +1,130 @@
+- What advantages could there be in random branching?
+- Why not use a less computationally intense algorithm like A*?
+# What a good question
+
+** Advantages of RRT
+- Specialized for continuous configuration spaces
+- Fast
+- Can handle kinodynamic constraints
+- Algorithm can be modified for various needs and preferences
+
+* RoboCup Modifications
+
+** Biases
+- Goal Bias
+  - Random branching has tendency to branch directly towards goal instead
+- Waypoint Bias
+  - Random branching has tendency to branch towards Bézier curve waypoints of previous paths
+- Goal Bias + Waypoint Bias must sum to at most 1.0
+
+** Adaptive Stepsize Control
+- Stepsize now dynamically changes based on whether there are obstacles nearby
+- Requires extra computation time to locate nearby obstacles
+- Having larger stepsizes when possible reduces total iteration count, which reduces computation time
+- Obstacle-light environments benefit the most from this enhancement
+
+** k-d Trees
+- Finding the node in the RRT to extend from is computationally expensive
+- Binary tree that attempts to maintain balance
+- Partitions the state space evenly based on the number of nodes in each partition
+- The branch that each node is stored under is determined by comparing that node to the median of the node in that subtree
+
+* STOx's Planner
+# http://fei.edu.br/rcs/2014/RegularPapers/robocupsymposium2014_submission_55.pdf
+# Columbian RoboCup team, developed this nameless algorithm in 2013
+- Generate a straight line from the start state to end state
+- As long as the path intersects an obstacle:
+  - Generate a subgoal state next to the obstacle
+  - Now divided into two smaller subproblems
+  - Recurse!
+
+* STOx's Planner
+[[file:http:/i.imgur.com/Ea040em.png]]
+
+* STOx's Planner
+- Very fast when obstacle count is low
+- Not very flexible
+
+* A*
+- Search space
+  - Discrete network of nodes
+  - Traversable edges between nodes
+- Generalized Dijkstra's algorithm
+  - Generalized breadth-first search
+
+* BFS
+[[file:https://upload.wikimedia.org/wikipedia/commons/4/46/Animated_BFS.gif]]
+
+* Dijkstra's algorithm
+- Like BFS, but acknowledges costs with each edge
+- Associates each edge with a distance cost, and assigns each node with a tentative distance cost
+# root starts at 0 distance, all other nodes start at infinity
+- At each iteration, update the distance to the nodes neighboring the current node
+- Select the unvisited node with the smallest tentative distance at the next iteration
+- BFS is Dijkstra's algorithm with equal edge weights
+
+* A*
+- Like Dijkstra's but with a heuristic function h(n)
+- Cost function f(n) = g(n) + h(n)
+  - g(n): cost of path from start to n
+  - h(n): estimate cost of cheapest path from n to goal
+- Heuristic must be admissible (no overestimating)
+
+* Dynamic A* Search
+- Searching backwards from goal to start
+- Efficient replanning and backtracking
+
+* Path Network
```

Comment:
Maybe (I'm not sure if you want to focus on this) you could talk about both grids and path networks here, since you can use any of the above algorithms with both of these strategies.